### PR TITLE
Added documentation. 

### PR DESCRIPTION
**Pull request by JamesDrummond**
### What does this PR do?
The markdown and configuration files are used by jekyll to generate static html. This static html can be used both in the eclipse che product with the already include tomcat server and on web hosting sites such as github.io .

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/3406

### Current generated website:  
http://45.33.14.150:84/docs/index.html

This can be move to che.github.io repo once it is created.

### CODE REVIEW

**Review by bmicklea — APPROVED** [review]

File: docs/_config.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+kramdown:
+  toc_levels:     1..6
+  smart_quotes:   lsquo,rsquo,ldquo,rdquo
+  input:          GFM
+#gauges_id: 503c5af6613f5d0f19000027
+#google_analytics_id: UA-50755011-1
+#google_site_verification: onQcXpAvtHBrUI5LlroHNE_FP0b2qvFyPq7VZw36iEY
+
+repository: https://github.com/codenvy/codenvy
```
Comment (benoitf) [inline docs/_config.yml:10]: repository is invalid, it's che there

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+highlighter: rouge
+
+
+kramdown:
+  toc_levels:     1..6
+  smart_quotes:   lsquo,rsquo,ldquo,rdquo
+  input:          GFM
+#gauges_id: 503c5af6613f5d0f19000027
+#google_analytics_id: UA-50755011-1
+#google_site_verification: onQcXpAvtHBrUI5LlroHNE_FP0b2qvFyPq7VZw36iEY
```
Comment (benoitf) [inline docs/_config.yml]: please remove these comments

File: docs/_config.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+collections:
+  docs:
+    output: true
+  tutorials:
+    output: true
+  posts:
+    permalink: /news/:year/:month/:day/:title/
+    output: true
+
+name: Codenvy Docs
```
Comment (benoitf) [inline docs/_config.yml]: invalid name

File: docs/_config.yml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+description: Cloud workspaces for development teams
+url: https://codenvy.com
```
Comment (benoitf) [inline docs/_config.yml]: invalid url

File: docs/_config.yml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+twitter:
+  username: codenvyhq
```
Comment (benoitf) [inline docs/_config.yml]: invalid username : @eclipse_che

File: docs/_config.yml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  -
+    scope:
+      path: "_docs/che-data-model"
+    values:
+      categories: [ "docs" , "data-model" ]
+  -
+    scope:
+      path: "_docs/artik-ide-plugin"
+    values:
+      categories: [ "artik" ]
```
Comment (benoitf) [inline docs/_config.yml:106]: missing newline

File: docs/_data/artik.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+- title: ARTIK IDE
```
Comment (benoitf) [inline docs/_data/artik.yml:1]: artik shouldn't be in artik repository ?

File: docs/_data/artik.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  - artik-config                                                                                                        
+  - artik-network                                                                                                       
+  - artik-usb-connection-setup                                                                                          
+  - artik-docker                                                                                                        
+  - artik-tutorial-blink-led                                                                                            
+  - artik-apisdk-management                                                                                             
+  - artik-api-docs                                                                                                      
+  - artik-release-notes                                                                                                 
+  - artik-wireless-setup                                                                                                
+  - artik-download-ide
```
Comment (benoitf) [inline docs/_data/artik.yml]: missing newline

File: docs/_data/docs.yml [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  - plugins-java-class-reference                                                                                                
+  - plugins-che-properties                                                                                                      
+  - plugins-native-access-to-the-workspace                                                                                      
+  - plugins-embed-htmljs                                                                                                        
+  - plugins-helloworld-extension                                                                                                
+  - plugins-themes 
+- title: CHE DATA MODEL
+  docs:
+  - che-data-model-stack
+  - che-data-model-samples
```
Comment (benoitf) [inline docs/_data/docs.yml]: missing newline

File: docs/_data/openshift.yml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+- title: OPENSHIFT PLUG-IN FOR ECLIPSE CHE
+  openshift:
+  - openshift-config
+  - openshift-user-guide
```
Comment (benoitf) [inline docs/_data/openshift.yml]: missing newline

File: docs/_docs/setup-alternatives/che-setupalt-docker-server.md [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+# Replace <path-for-data> with any host folder
+# Che will place backup files there - configurable properties, workspaces, lib, storage
+docker run -p 8080:8080 \
+           --name che \
+           --rm \
+           -v /var/run/docker.sock:/var/run/docker.sock \
+           -v <path-for-data>:/data \
+           eclipse/che-server:5.0.0-latest
+
+# To run the nightly version of Che, replace codenvy/che with
```
Comment (benoitf) [inline docs/_docs/setup-alternatives/che-setupalt-docker-server.md]: codenvy/che ? seems not ok

File: docs/_docs/setup-alternatives/che-setupalt-docker-server.md [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+eclipse/che-server:nightly
+
+# To run a specific tagged version of Che, replace codenvy/che with
```
Comment (benoitf) [inline docs/_docs/setup-alternatives/che-setupalt-docker-server.md]: codenvy/che ?

File: docs/dev/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2012-2016 Codenvy, S.A.
```
Comment (benoitf) [inline docs/dev/Dockerfile]: what is the purpose of that dockerfile ?
now Dockerfile are in /dockerfiles folder

File: docs/dev/default.prf [hunk 14]
```diff
@@ -0,0 +1,23 @@
+# Arguments
```
Comment (benoitf) [inline docs/dev/default.prf]: what is this file ? 
there is no copyright

File: docs/dev/docker-compose.yml [hunk 15]
```diff
@@ -0,0 +1,9 @@
+version: '2'
+services:
+  docs:
+    build: .
+    ports:
+      - "${JEKYLL_BIND_PORT}4000"
+      - "22"
+    container_name: ${CONTAINER_NAME}
+    network_mode: "bridge"
```
Comment (benoitf) [inline docs/dev/docker-compose.yml]: missing newline
dockerfiles are in /dockerfiles folder in che

File: docs/dev/jekyll.sh [hunk 16]
```diff
@@ -0,0 +1,217 @@
+#!/bin/sh
```
Comment (benoitf) [inline docs/dev/jekyll.sh]: what is this file ?

File: docs/pom.xml [hunk 17]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+     [2012] - [2016] Codenvy, S.A.
```
Comment (benoitf) [inline docs/pom.xml]: I think license is invalid there it should be EPL

File: docs/pom.xml [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+    patents in process, and are protected by trade secret or copyright law.
+    Dissemination of this information or reproduction of this material
+    is strictly forbidden unless prior written permission is obtained
+    from Codenvy S.A..
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>onpremises-assembly-parent</artifactId>
```
Comment (benoitf) [inline docs/pom.xml]: the parent is invalid

**Review by benoitf — CHANGES_REQUESTED** [review]

**Comment by benoitf** [comment]
```
docs/assets/fonts/FontAwesome.eot
docs/assets/fonts/FontAwesome.svg
docs/assets/fonts/FontAwesome.ttf
docs/assets/fonts/FontAwesome.woff
```

what is the version of FontAwesome ? is it one for which we have CQs ?

File: docs/_config.yml (hunk 2 above)
Comment (JamesDrummond) [reply docs/_config.yml]: removed

File: docs/_config.yml (hunk 1 above)
Comment (JamesDrummond) [reply docs/_config.yml:10]: updated

File: docs/_config.yml (hunk 3 above)
Comment (JamesDrummond) [reply docs/_config.yml]: updated

File: docs/_config.yml (hunk 4 above)
Comment (JamesDrummond) [reply docs/_config.yml]: Updated

File: docs/_config.yml (hunk 5 above)
Comment (JamesDrummond) [reply docs/_config.yml]: Updated

File: docs/_data/docs.yml (hunk 9 above)
Comment (JamesDrummond) [reply docs/_data/docs.yml]: Added.

File: docs/_data/openshift.yml (hunk 10 above)
Comment (JamesDrummond) [reply docs/_data/openshift.yml]: Added.

File: docs/_docs/setup-alternatives/che-setupalt-docker-server.md (hunk 11 above)
Comment (JamesDrummond) [reply docs/_docs/setup-alternatives/che-setupalt-docker-server.md]: Replaced

File: docs/_docs/setup-alternatives/che-setupalt-docker-server.md (hunk 12 above)
Comment (JamesDrummond) [reply docs/_docs/setup-alternatives/che-setupalt-docker-server.md]: Replaced.

**Comment by JamesDrummond** [comment]
@benoitf Please see changes in new commit.

**Comment by JamesDrummond** [comment]
Review my repo if viewing PR is difficult due to the number of files. Folder /docs content is all new. https://github.com/JamesDrummond/che/tree/jekyll-docs/docs versus master.

**Review by TylerJewell — APPROVED** [review]

**Comment by benoitf** [comment]
https://github.com/eclipse/che/blob/master/docs/assembly/build_war.yml has reference to codenvy
Also pr was merged while I was still requesting changes

**Pull request merged**